src/tests/dump/test_to_db.py

```diff
@@ -431,6 +431,109 @@ def test_dump_to_db_partial_no_source_ids_to_update(self, mock_get_entity_refs,
+        mock_model.return_value = MockedModel()
+        db_dumper = DbDumper('catalog', 'collection', {'db': {}})
+        db_dumper.model = {
+            'abbreviation': 'abbr',
+            'has_states': False,
+            'references': {
+                'refA': {},
+                'refB': {}
+            },
+            'all_fields': {
+                'refA': {
+                    'ref': 'dstcat:dstcolwithstates',
+                    'type': 'GOB.ManyReference',
+                },
+                'refB': {
+                    'ref': 'dstcat:dstcol',
+                    'type': 'GOB.Reference',
+                }
+            }
+        }
+
+        expected_query = """create view v_catalog_collection as 
+select abbr.*,
+       refA.ref refA_ref,
+       refA.dst_id refA_id,
+       refA.bronwaarde refA_bronwaarde,
+       refA.dst_volgnummer refA_volgnummer,
+       refB.dst_id refB_ref,
+       refB.dst_id refB_id,
+       refB.bronwaarde refB_bronwaarde
+from catalog_collection abbr
+
+left join (
+    select
+        rel.src_id,
+        array_agg(rel.dst_id) dst_id,
+        rel.dst_volgnummer dst_volgnummer,
+        array_agg(rel.bronwaarde) bronwaarde,
+        array_agg(rel.dst_id || \'_\' || rel.dst_volgnummer) "ref"
+    from rel_relation_name_refA rel
+    where rel._date_deleted is null
+    group by rel.src_id
+) refA on refA.src_id = abbr._id
+
+left join rel_relation_name_refB refB on refB.src_id = abbr._id and refB._date_deleted is null
+"""
+
+        self.assertEqual(['Utility view v_catalog_collection created\n'], list(db_dumper.create_utility_view()))
+        db_dumper.engine.execute.assert_has_calls([
+            call('drop view if exists v_catalog_collection'),
+            call(expected_query)
+        ])
+
+        # Second test case. src has states, refA single Reference and refB a ManyReference
+        db_dumper.model['has_states'] = True
+        db_dumper.model['all_fields']['refA']['type'] = 'GOB.Reference'
+        db_dumper.model['all_fields']['refB']['type'] = 'GOB.ManyReference'
+        db_dumper.engine.execute.reset_mock()
+
+        expected_query = """create view v_catalog_collection as 
+select abbr.*,
+       refA.dst_id || '_' || refA.dst_volgnummer refA_ref,
+       refA.dst_id refA_id,
+       refA.bronwaarde refA_bronwaarde,
+       refA.dst_volgnummer refA_volgnummer,
+       refB.ref refB_ref,
+       refB.dst_id refB_id,
+       refB.bronwaarde refB_bronwaarde
+from catalog_collection abbr
+left join rel_relation_name_refA refA on refA.src_id = abbr._id and refA.src_volgnummer = abbr.volgnummer and refA._date_deleted is null
+
+left join (
+    select
+        rel.src_id,
+        array_agg(rel.dst_id) dst_id,
+        
+        array_agg(rel.bronwaarde) bronwaarde,
+        array_agg(rel.dst_id) "ref"
+    from rel_relation_name_refB rel
+    where rel._date_deleted is null
+    group by rel.src_id
+) refB on refB.src_id = abbr._id and refB.src_volgnummer = abbr.volgnummer
+
+"""
```

Comment:
I suggest to make a compare_query method oid, that compares a query while removing comments. Someone can the add comments to queries to make them more readable withour having to update tests. Only the logic should be compared not the layout

Reply:
Done. Also added whitespace stripping to really ignore layout (as done elsewhere)